### DISCUSSION TITLE
Feature request: Wake up portable speakers (Roam, Move etc.)

### DISCUSSION THREAD

**Discussion by anes300** [punxaphil/custom-sonos-card#132] [Show and tell]
Hi Johan,

First of all, I want to thank you for the great work you've put into this Sonos Card. The newest version really lifts up the UI of the card making it pleasing to use at home.

I have a request which might not be directly related to the card but could be for some pepole with battery-powers portable speakers. So, I've got a Sonos Roam in the bathroom which goes into sleep mode after 30 minutes. From here, it's not available to the Sonos Card or Home Assistant in general leaving me with two options: either opening the Sonos app from my phone which wakes up the speaker or sending a WOL packet from Home Assistant through a script/service.

Normally, I would just add a button next to your card which calls the WOL script, but I would very much hope for an integration directly within your card. This would for me make most sense to be part of either the groups or grouping section.

Let me know what you think of this idea, and once again thank you for your great work!

Sincerely,
Anes Skrijelj

**Comment by punxaphil** [comment]
Thanks for the suggestion, and I understand it could be nice. I'll add it to my list, but I'm afraid there are quite a few fixes and features that are higher prio, so no promises 🙂 
Meanwhile, feel free to share more details on the script/service needed.

**Comment by anes300** [comment]
Hi again,

Totally understandable - this was more an idea for the future if you get the time. 😊 What I've done in the meantime is to add a button to the top of the screen which is only shown if the speaker is unavailable (in sleep-mode). Then I've used the same condition to strecth out your card in height to ensure the screen's height is used out:

<img width="524" alt="image" src="https://github.com/johanfrick/custom-sonos-card/assets/54175142/a97a2836-eebf-47be-9786-0914b29250e8">

The code for the button looks like this, and I'm a bit unsure how this would be handled within the card as I'm not entirely sure that it's possible to retrieve the MAC-address from the integration. This would then have to be imported/typed manually.

```
service: wake_on_lan.send_magic_packet
data:
  broadcast_port: 9
  mac: XX:XX:XX:XX:97:1C
```

The solution above works fine for my needs, and the WOL-method is very fast, making the speaker available to the card within a second. Since this is also an uncommon scenario for most (who probably use nonBattery-powered speakers for HA), I'll leave this here as an idea.

Sincerely,
Anes Skrijelj

**Comment by punxaphil** [comment]
Thanks for sharing, Anes. Let's keep it here as a "Show and tell" under discussions instead. So anyone else needing this can use your example 😄

**Reply by anes300** [reply]
Sounds good to me! 😊
